src/pip/_internal/index/package_finder.py

```diff
@@ -709,6 +709,15 @@ def set_allow_all_prereleases(self):
         # type: () -> None
         self._candidate_prefs.allow_all_prereleases = True
 
+    @property
+    def prefer_binary(self):
+        # type: () -> bool
+        return self._candidate_prefs.prefer_binary
```

Comment:
This isn't necessary.

Reply:
I am using it here: https://github.com/pypa/pip/pull/7996/files#diff-03a81e5b9407d12be1f9457a7925d422R196 in file `src/pip/_internal/build_env.py` to get the value of `self._candidate_prefs.prefer_binary`

For more context, I am following the same pattern used to expose `self._candidate_prefs.allow_all_prereleases` as a property

https://github.com/pypa/pip/blob/daff81124ab76138da323a5173b120dd88190f56/src/pip/_internal/index/package_finder.py#L703-L710

We are using the property getter for it here:

https://github.com/pypa/pip/blob/daff81124ab76138da323a5173b120dd88190f56/src/pip/_internal/build_env.py#L194-L195

And the setter here.

https://github.com/pypa/pip/blob/daff81124ab76138da323a5173b120dd88190f56/src/pip/_internal/req/req_file.py#L260-L261

Is there an alternative to this approach?

Reply:
Ah! I missed that in a hurry. Whoops!